src/include/btree.i

```diff
@@ -1424,12 +1424,11 @@ __wt_page_evict_retry(WT_SESSION_IMPL *session, WT_PAGE *page)
       mod->last_eviction_id != __wt_txn_oldest_id(session))
         return (true);
 
-    if (mod->last_eviction_timestamp == WT_TS_NONE)
-        return (true);
-
-    __wt_txn_pinned_timestamp(session, &pinned_ts);
-    if (pinned_ts > mod->last_eviction_timestamp)
-        return (true);
+    if (mod->last_eviction_timestamp != WT_TS_NONE) {
```

Comment:
This looks as expected, but I think it deserves a comment describing why the clause is structured this way.